### DISCUSSION TITLE
Validation for optional field?

### DISCUSSION THREAD

**Discussion by jonmor51** [quasarframework/quasar#11307] [General - Components / Directives / etc]
I've been using Vue for quite a while now but I'm a beginner with Quasar and I'm just getting my head round certain things. I like the simplicity of the inbuilt validation system, and I have worked out now how to use it with my own validation rules defined in methods of the Vue instance. One thing I'm not doing right, though, is validation for an optional field. I have a form in which there is an 'email' input. It isn't mandatory, but if the user enters anything into it, it needs to be a valid email address. So how do I enforce that? My email rule at the moment is this:

`const emailRegex=/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
return this.customer.email && emailRegex.test(this.customer.email) || 'Please enter a valid email address';`

But that is flagging a validation error if the user enters something in the customer field and then deletes it.

**Comment by MilosPaunovic** [comment]
Check `lazy-rules` prop of `QInput`.

> If set to boolean true then it checks validation status against the 'rules' only after field loses focus for first time; If set to 'ondemand' then it will trigger only when component's validate() method is manually called or when the wrapper QForm submits itself

[https://quasar.dev/vue-components/input#qinput-api](https://quasar.dev/vue-components/input#qinput-api)

**Reply by jonmor51** [reply]
I believe I have found the answer, understanding the logic a little better. I should have had this instead:

`return (!this.customer.email || emailRegex.test(this.customer.email)) || 'Please enter a valid email address';`

I.e., it's valid if there's nothing in the field or it matches the regex, otherwise display the message. So my understanding is that I can have any number of and/or conditions together, with the text after the last || being displayed if the validation fails? So I could have `a || b || (c &&d) || e || 'Validation failed'`. Is this correct? (Of course, I could have the empty string built in as an option in the regex, which might be a better approach, but I think it is clearer the way I have it).
